custom_components/smartthinq_sensors/binary_sensor.py

```diff
@@ -343,18 +342,12 @@ def assumed_state(self) -> bool:
         """Return True if unable to access real state of the entity."""
         return self._api.assumed_state
 
-    def _get_on_state(self):
+    def _get_on_state(self) -> bool:
         """Return true if the binary sensor is on."""
-        ret_val = self._get_sensor_state()
-        if ret_val is None:
```

Comment:
If we don't manage the None condition, the .`lower()` function below could raise an exception.